src/Stratis.Bitcoin.Features.Miner/PowMining.cs

```diff
@@ -196,6 +196,12 @@ private bool ConsensusIsAtTip(MineBlockContext context)
             this.miningCancellationTokenSource.Token.ThrowIfCancellationRequested();
 
             context.ChainTip = this.consensusManager.Tip;
+
+            // Genesis on a regtest network is a special case. We need to regard ourselves as outside of IBD to
+            // bootstrap the mining.
+            if (this.network.IsRegTest() && (context.ChainTip.Height == 0))
```

Comment:
hmmm I wonder what is the problem here exactly?
Genesis is always a special case o maybe this can be solved without  needing to check RegTest

Reply:
The original issue says `this.ChainTip is null` perhaps we just need to return if its null.

Reply:
The specific problem is an infinite loop in the IBD check, because we were regarding ourselves as always in IBD at genesis. For most chains this is technically a reasonable assumption, it is only really with regtest networks that we are constantly generating entirely new chains from scratch.

I suppose the same issue would occur with chains mined from scratch on any network, so perhaps the `IsRegTest` portion of the check should just be removed. Thoughts?

Reply:
With regard to the original issue, I think that investigation is extremely out of date now. Perhaps there is some merit to checking the null-ness of ChainTip, but the check I added definitely appears to fix the problem, and I don't see the same symptoms manifesting that I originally observed.

Reply:
yeah remove regtest and I think it will work

Reply:
Done